lib/Doctrine/ORM/Persisters/Collection/OneToManyPersister.php

```diff
@@ -163,7 +164,20 @@ public function removeElement(PersistentCollection $collection, $element)
         $mapping   = $collection->getMapping();
         $persister = $this->uow->getEntityPersister($mapping['targetEntity']);
 
-        return $persister->delete($element);
+        $targetMetadata = $this->em->getClassMetadata($mapping['targetEntity']);
+
+        if ($element instanceof Proxy && ! $element->__isInitialized()) {
+            $element->__load();
+        }
+
+        // clearing owning side value
+        $targetMetadata->reflFields[$mapping['mappedBy']]->setValue($element, null);
```

Comment:
This allows the "inverse" side to change the "owning-side", which is exactly what we didn't want to do.